services/proxy/compose-utils.go

```diff
@@ -49,10 +58,51 @@ func buildNetworks(networks []string) []byte {
 			continue
 		}
 
+		// Verify that this network still exists in Docker
+		ok, err := validateNetwork(v)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if !ok {
+			// That network no longer exists, don't add it
+			utils.DebugString("network [" + v + "] no longer exists, not adding it to proxy service")
+			continue
+		}
+
 		n = n + `  ` + strings.Replace(v, ".", "", -1) + `:
     external: true
 `
 	}
 
 	return []byte(n)
 }
+
+// validateNetwork confirms that a docker network exists
+// and returns false if no match is found
+func validateNetwork(name string) (ok bool, err error) {
+	dcli := docker.GetAPIClient()
+
+	filter := filters.NewArgs()
+	filter.Add("name", name)
+
+	networks, err := dcli.NetworkList(context.Background(), types.NetworkListOptions{
+		Filters: filter,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	if len(networks) == 1 {
+		return true, nil
+	}
+
+	if len(networks) == 0 {
+		return false, nil
+	}
+
+	if len(networks) > 1 {
+		return false, fmt.Errorf("error looking up network [%s]. Received [%d] matching networks when only expecting 1 or none", name, len(networks))
```

Comment:
Good edge catch here, my feel is that we need to keep an ear/eye out for this actually happening. Stale networks, messed up state happens in docker, wouldn't want this occurring too often for our users

Reply:
yeah - at some point I'd like to add a `tok cleanup` command that just goes through and removes all tokaido environments, config, networks, volumes, etc and resets back to zero for those rare cases where things get completely messed up